galley/pkg/testing/kubernetes/mock/corev1.go

```diff
@@ -0,0 +1,114 @@
+//  Copyright 2018 Istio Authors
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package mock
+
+import (
+	corev1 "k8s.io/client-go/kubernetes/typed/core/v1"
```

Comment:
Do we have to use the typed clients? Using dynamic client may make the mock easy to setup and less dependent to k8s apis?